### PR TITLE
Technical fixes for geometry DB payload creation

### PR DESCRIPTION
**Pull request by cvuosalo**
This PR is a collection of fixes and enhancements to support geometry DB payload creation. The changes are as follows.

`CondTools/Geometry/test/writehelpers/geometryExtended2021_writer.py` -- remove improper DD4hep dependence from this DDD script.

`DetectorDescription/OfflineDBLoader/plugins/OutputDDToDDL.cc` -- fix to prevent run-time error related to Event Setup.

`Geometry/CSCGeometry/test/stubs/CSCGeometryAnalyzer.cc` -- ES Get Token migration to allow CSC geometry test to run in 12_3.

`Geometry/GEMGeometry/test/testGEMGeometry_cfg.py`
`Geometry/GEMGeometry/test/testGEMGeometryFromDB_cfg.py`
`Geometry/GEMGeometry/test/testGEMGeometryFromLocalDB_cfg.py` -- scripts to help compare GEM reco geometry in XML, DB, and a local DB.

None of the files in this PR are used by any workflow.  They are only used for expert tools.

#### PR validation:

DDD DB payloads were created succesfully to verify the changes in this PR. Also, the test scripts were run, and they produce reasonable output.

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

A backport to 12_2 would promote consistency but it is not strictly necessary. I can manually copy these changes to a 12_2 working directory if it turns out I need them.

### CODE REVIEW

File: Geometry/CSCGeometry/test/stubs/CSCGeometryAnalyzer.cc [hunk 1]
```diff
@@ -27,10 +27,13 @@ class CSCGeometryAnalyzer : public edm::one::EDAnalyzer<> {
   const int dashedLineWidth_;
   const std::string dashedLine_;
   const std::string myName_;
+  edm::ESGetToken<CSCGeometry, MuonGeometryRecord> ddToken_;
```
Comment (tvami) [inline Geometry/CSCGeometry/test/stubs/CSCGeometryAnalyzer.cc:30]:
```suggestion
  const edm::ESGetToken<CSCGeometry, MuonGeometryRecord> ddToken_;
```

File: Geometry/CSCGeometry/test/stubs/CSCGeometryAnalyzer.cc [hunk 2]
```diff
@@ -42,8 +45,7 @@ void CSCGeometryAnalyzer::analyze(const edm::Event& iEvent, const edm::EventSetu
   std::cout << "start " << dashedLine_ << std::endl;
   std::cout << "pi = " << dPi << ", radToDeg = " << radToDeg << std::endl;
 
-  edm::ESHandle<CSCGeometry> pDD;
-  iSetup.get<MuonGeometryRecord>().get(pDD);
+  edm::ESTransientHandle<CSCGeometry> pDD = iSetup.getTransientHandle(ddToken_);
```
Comment (tvami) [inline Geometry/CSCGeometry/test/stubs/CSCGeometryAnalyzer.cc:48]: I think this can be `const` too

Comment (cvuosalo) [reply Geometry/CSCGeometry/test/stubs/CSCGeometryAnalyzer.cc]: @tvami Thanks for the suggestions. I made those changes.

**Comment by cvuosalo** [comment]
@cmsbuild please test

**Comment by civanch** [comment]
+1

**Comment by tvami** [comment]
I see the vertex collection changed for 2026D88
https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_3_X_2022-02-04-2300+dd7f16/48241/39496.0_CloseByPGun_CE_E_Front_120um+2026D88+CE_E_Front_120um_GenSimHLBeamSpotHGCALCloseBy+DigiTrigger+RecoGlobal+HARVESTGlobal/MTD_Vertices.html
that cannot be connected to this PR, right?

**Comment by tvami** [comment]
+db
 * the comment here https://github.com/cms-sw/cmssw/pull/36897#issuecomment-1030697531 should be confirmed
 * for the future the test scripts could also be made a unit test

**Comment by srimanob** [comment]
+Upgrade

Technical PR for payload creation.

**Comment by perrotta** [comment]
+1
- Updates made on some private script
- (@tvami those non reproducibilities in the MVT/Vertex DQM plots appears in all PR test comparisons since some while: it is certainly worth identifying their origin, which is not in this PR, anyhow)

**Pull request merged**